packages/modals/src/elements/DrawerModal/DrawerModal.tsx

```diff
@@ -49,12 +59,37 @@ const DrawerModalComponent = forwardRef<HTMLDivElement, IDrawerModalProps>(
       useModal({
         idPrefix: id,
         modalRef,
-        focusOnMount,
-        restoreFocus,
+        focusOnMount: false /* [1:b] */,
+        restoreFocus: false /* [1:b] */,
         environment,
         onClose
       });
 
+    /* [1:c] */
+    useEffect(() => {
+      if (environment) {
+        if (isOpen && modalRef.current) {
+          if (restoreFocus) {
+            triggerRef.current = activeElement(environment) as HTMLElement;
+          }
+
+          if (focusOnMount) {
+            modalRef.current.focus();
+          }
+        }
+
+        if (!isOpen && triggerRef.current) {
+          triggerRef.current.focus();
+        }
+      }
+
+      return () => {
+        if (!(restoreFocus && isOpen)) {
+          triggerRef.current = null;
+        }
+      };
+    }, [environment, restoreFocus, focusOnMount, isOpen]);
```

Comment:
I'm curious, why was `useFocusJail` not enhanced accordingly? It seems like the logic would be more beneficial down there.

Reply:
That was the initial intention and as we discussed offline, that is definitely a future goal.